src/server/api/lib/message-sending.js

```diff
@@ -1,4 +1,222 @@
 import { r } from "../../models";
+import { config } from "../../../config";
+
+export const SpokeSendStatus = Object.freeze({
+  Queued: "QUEUED",
+  Sending: "SENDING",
+  Sent: "SENT",
+  Delivered: "DELIVERED",
+  Error: "ERROR",
+  Paused: "PAUSED",
+  NotAttempted: "NOT_ATTEMPTED"
+});
+
+/**
+ * Return a list of messaing services for an organization that are candidates for assignment.
+ *
+ * TODO: Update logic to allow for per-campaign decisions.
+ *
+ * @param {number} organizationId The ID of organization
+ */
+export const getMessagingServiceCandidates = async organizationId => {
+  const { rows: messagingServiceCandidates } = await r.knex.raw(
+    `
+      select
+        messaging_service.messaging_service_sid,
+        count(messaging_service_stick.messaging_service_sid) as count
+      from messaging_service
+      left join messaging_service_stick
+        on messaging_service_stick.messaging_service_sid = messaging_service.messaging_service_sid
+      where
+        messaging_service.organization_id = ?
+      group by
+        messaging_service.messaging_service_sid
+      order by count desc
+    `,
+    [organizationId]
+  );
+  return messagingServiceCandidates;
+};
+
+/**
+ * Assign an appropriate messaging service for a (cell, organization) pairing.
+ * This creates a messaging_service_stick record.
+ *
+ * TODO: Update logic to allow for per-campaign decisions.
+ *
+ * @param {string} cell An E164-formatted destination cell phone number
+ * @param {number} organizationId The ID of the organization to create the mapping for
+ * @returns {string} The (S)ID of the messaging service assigned to that (cell, organization)
+ */
+export const assignMessagingServiceSID = async (cell, organizationId) => {
+  const {
+    rows: [{ messaging_service_sid }]
+  } = await r.knex.raw(
+    `
+      with chosen_messaging_service_sid as (
+        select
+          messaging_service.messaging_service_sid,
```

Comment:
```suggestion
          messaging_service.*,
```

return everything to avoid another round trip!